processor/src/main/resources/org/mapstruct/ap/internal/model/BeanMappingMethod.ftl

```diff
@@ -24,6 +24,19 @@
     }
     </#if>
 
+    <#if hasSubClassMappings()>
+        <#assign first = true />
+        <#list constructSubClassMappings(parameters) as subClass>
+            <#if !first>else</#if> if (${subClass.sourceArgument} instanceof <@includeModel object=subClass.sourceType/>) {
+                <@includeModel object=subClass.assignment existingInstanceMapping=existingInstanceMapping/>
+            }
+            <#assign first = false />
+        </#list>
+        else {
+    </#if>
+    <#if isAbstractReturnType()>
+        throw new IllegalArgumentException("Not all subclasses are supported for this mapping. Missing for " + ${parameters[0].name}.getClass());
```

Comment:
changed it so it uses the same value as the subclassmappings themselves do by getting the parameter name from the first subclassmapping. That way it will be the correct parameter.

On another note, no idea how to find the correct method in the following case:
```java
@SubClassMapping(....)
@Mapping(...., source="otherSource.value")
Target mapSuperClass(SourceSuperClass source, OtherSource otherSource);

@Mapping(....)
Target mapSubClass(SourceSubClass source, OtherSource otherSource);
```
didn't write out the whole thing, but my primary tests show that I'll generate a new method.
`Target mapSourceSubClassToTarget(SourceSubClass source)`
Instead of using the method `mapSubClass` with the additional parameter.

Although my suggestion to those who need this will be create a record containing those 2 arguments so that you only have to pass 1 argument. That way mapstruct will be able to handle it correctly.

Reply:
Multi argument methods are not reused on our side. So you are correct not to focus on that.